app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  class RepositoryTypeManager
+    @repository_types = {}
+    class << self
+      attr_reader :repository_types
+      private :new
+      # Plugin constructor
+      def register(id, &block)
+        repository_type = ::Katello::RepositoryType.new(id)
+        repository_type.instance_eval(&block)
+        repository_types[id.to_s] = repository_type
+      end
+
+      def creatable_repository_types
+        repository_types.select do |repo_type, _|
```

Comment:
Could you drop the second parameter that is unused? Or switch to `repository_types.keys.select` ?